scripts/daq_utils.py

```diff
@@ -15,185 +15,212 @@
-                    "state": state,
-                    "nodelist": nodelist,
-                    "logfile": logfile,
-                }
+                for scontrol_line in scontrol_output.splitlines():
+                    if "StdOut" in scontrol_line:
+                        parts = scontrol_line.split("=")
+                        if len(parts) > 1:
+                            logfile = parts[1]
+            except Exception as e:
+                logging.error("Failed to get scontrol info for job %s: %s", job_id, e)
+                logfile = ""
+
+            job_details[job_name] = {
+                "job_id": job_id,
+                "comment": comment,
+                "state": state,
+                "nodelist": nodelist,
+                "logfile": logfile,
+            }
         return job_details
 
 
 class DaqManager:
     def __init__(self, verbose=False, cnf=None):
         self.verbose = verbose
-        self.hutch = call_subprocess("get_info", "--gethutch")
+        try:
+            self.hutch = call_subprocess("get_info", "--gethutch")
+        except Exception as e:
+            logging.error("Failed to get hutch info: %s", e)
+            raise
         if len(self.hutch) != 3:
             raise ValueError(f"Invalid hutch name found (hutch: '{self.hutch}')")
 
         self.user = self.hutch + "opr"
         self.sbman = SbatchManager(self.user)
         self.scripts_dir = f"/reg/g/pcds/dist/pds/{self.hutch}/scripts"
-        if cnf is None:
-            self.cnf_file = f"{self.hutch}.py"
-        else:
-            self.cnf_file = cnf
-
-    def isdaqmgr(self, quiet=False):
-        if self.hutch in DAQMGR_HUTCHES:
-            if not quiet:
-                print("true")
-            return True
-        else:
-            if not quiet:
-                print("false")
-            return False
+        self.cnf_file = f"{self.hutch}.py" if cnf is None else cnf
+        logging.debug(
+            "DaqManager initialized for hutch %s with config %s",
+            self.hutch,
+            self.cnf_file,
+        )
 
     def isvaliduser(self):
-        return getpass.getuser() == self.user
+        valid = getpass.getuser() == self.user
+        if not valid:
+            logging.error(
+                "Invalid user. Expected %s but got %s", self.user, getpass.getuser()
+            )
+        return valid
 
     def waitfor(self, subcmd):
         if subcmd == "stop":
-            daq_host = self.wheredaq(quiet=True)
+            daq_host = self.wheredaq_internal()
             if daq_host is not None:
-                if self.isdaqmgr(quiet=True):
+                if self.hutch in DAQMGR_HUTCHES:
                     for i_retry in range(MAX_RETRIES):
-                        daq_host = self.wheredaq(quiet=True)
+                        daq_host = self.wheredaq_internal()
                         if daq_host is None:
+                            logging.debug("DAQ has stopped after %d retries.", i_retry)
```

Comment:
Note for the future: It's probably totally good for now, but if we end up adding even more if/indent chains here it gets kind of hard to read (I blame Python!), so in the future we should consider breaking some of these bits out into helper functions to keep it readable.